### PR TITLE
add flake.nix

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by Mic92** [comment]
Have not test in depth yet... basic import seems to work.

**Comment by Mic92** [comment]
Works now. I have uploaded an example here:

https://git.thalheim.io/Mic92/kubes/src/commit/6f3a8542c81f524c162ce8ed52643ebe4d965b40/flake.nix#L4

**Comment by Mic92** [comment]
I would add an example to the README once I get some feedback on the PR.

**Comment by blaggacao** [comment]
I'd love to see this get merged rather sooner than later. @offlinehacker Any chances?

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  description = "Kubernetes resource builder using nix";
+
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs";
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system}; in
+      rec {
+        packages.kubenix = pkgs.callPackage ./default.nix {
```
Comment (offlinehacker) [inline flake.nix:11]: maybe better to use `lib` instead of `packages`?

Comment (blaggacao) [reply flake.nix:11]: `packages` is the one of the [flake special outputs](https://github.com/NixOS/nix/blob/master/src/nix/flake-check.md#evaluation-checks), so conforming to this convention this allegedly is required for `nix run ...` to work.

Comment (offlinehacker) [reply flake.nix:11]: ok

**Pull request merged**